Dark mode + UI fixes

mayur1377 opened the pull request:
https://github.com/RequestNetwork/easy-invoice/issues/46 : add dark mode , a few snapshots below

<img width="1680" height="926" alt="Screenshot 2025-08-01 at 12 38 17 AM" src="https://github.com/user-attachments/assets/e5006c46-2550-4427-8f4b-a58251899db9" />
<img width="1660" height="927" alt="Screenshot 2025-08-01 at 12 38 41 AM" src="https://github.com/user-attachments/assets/722e94b1-bcf6-45b4-b68a-63aadfb172d9" />
<img width="860" height="213" alt="Screenshot 2025-08-01 at 12 40 14 AM" src="https://github.com/user-attachments/assets/6d8a09d4-6544-4d87-8f6f-aee40688bf53" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - System-aware dark mode with a visible theme toggle.
  - User menu adds a "Copy email" action.
  - Theme provider expanded to cover more UI for consistent theming.

- Style
  - Wide migration to semantic design tokens for colors, borders, and hover states.
  - Updated visuals for buttons, links, cards, tables, badges, tooltips, loaders, and empty states.
  - Standardized success/warning/destructive/muted/foreground styles across components.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mayur1377 committed:
feat: add dark mode

MantisClone commented:
Hello @mayur1377, thank you for submitting your first pull request to the easy-invoice repository. We value your contribution and encourage you to review our contribution guidelines to ensure your submission meets our standards. Please note that every merged PR is automatically enrolled in our Best PR Initiative, offering a chance to win $500 each quarter. Our team is available via [GitHub Discussions](https://github.com/orgs/RequestNetwork/discussions) or [Discord](https://request.network/discord) if you have any questions. Welcome aboard!

MantisClone commented:
Thank you for your submission! As you prepare for the review process, please ensure that your PR title, description, and any linked issues fully comply with our contribution guidelines. A clear explanation of your changes and their context will help expedite the review process. Every merged PR is automatically entered into our Best PR Initiative, offering a chance to win $500 every quarter. We appreciate your attention to detail and look forward to reviewing your contribution!

bassgeta commented on src/components/direct-payout.tsx:
```diff
@@ -229,20 +233,22 @@ export function DirectPayment() {
-                          ? "border-black bg-zinc-50"
-                          : "border-zinc-300"
+                          ? "border-foreground bg-muted"
+                          : "border-border"
```
Is it possible to rename border-border into border-primary or something? `border-border` is a bit funny :D

bassgeta reviewed (changes requested):
Thank you for your contribution to Easy Invoice <3
Really good work, thanks for switching our tailwind variable names to something semantic instead of just using the colour names...

On top of the 3 things I commented on, I've noticed a pretty weird bug which happens only when loading the app the very first time:
<img width="2401" height="1265" alt="image" src="https://github.com/user-attachments/assets/9e28ff1e-4dea-4487-a967-15d5b0822428" />
Something gets mixed up and some components end up in their dark theme variant, while others do not.
Tried this out on multiple Chromium browsers and I can reproduce it constantly.

The error goes away after toggling the theme at least once.

Not sure what the solution could be, but maybe the theme switcher should check if theme isn't yet set, to just default to the light theme.

Other than that loving the dark mode look :sunglasses: 
<img width="2325" height="1218" alt="image" src="https://github.com/user-attachments/assets/d796f309-d6b8-4843-96fe-d0f437dae03f" />

Some more issues with dark mode:
1. Invoice me links
<img width="2325" height="779" alt="image" src="https://github.com/user-attachments/assets/5cd438cc-42aa-4747-b5ac-ad7cb8bca129" />
2. The alert modal's delete text is dark, should probably be white
<img width="1273" height="893" alt="image" src="https://github.com/user-attachments/assets/05122caa-e350-42a9-bfa0-3d4be8518b50" />
3. The user popup is still white in dark mode
<img width="1460" height="351" alt="image" src="https://github.com/user-attachments/assets/ceea0cb8-4bda-476e-acfa-380106032210" />
4. The `/invoices/:id` page is borked
<img width="2429" height="1557" alt="image" src="https://github.com/user-attachments/assets/bfb74637-4187-4304-bc71-07d87021766d" />

And sadly, mandatory merge conflicts due to one of our huge PRs coming in, sorry about that :sweat_smile: 
After merging, do check the `/subscription-plans` route, there's a ton of added things there now.

After the UI issues and review comments are taken care of, I'll gladly drop an approval :100:

bassgeta reviewed (changes requested):
Hey! Thanks for resolving the merge conflicts, sadly some new ones appeared :sweat_smile: 
Additionally, I can still reproduce all the bugs I've pointed out in my previous review, can we take care of those too? :pray: 
Feel free to gimme a shout if you need some help or pointers!

mayur1377 commented:
> Hey! Thanks for resolving the merge conflicts, sadly some new ones appeared 😅 Additionally, I can still reproduce all the bugs I've pointed out in my previous review, can we take care of those too? 🙏 Feel free to gimme a shout if you need some help or pointers!

it would be very much helpful if you could resolve the conflicts ^^

mayur1377 committed:
feat: add dark mode

bassgeta reviewed (changes requested):
Alright thanks a bunch for the updates! Just got around to checking this PR out again.

Let's go over the issues I've pointed out last time first:
1. Invoice me links are now correctly styled :heavy_check_mark: 
<img width="1120" height="703" alt="image" src="https://github.com/user-attachments/assets/2a532c59-ed49-4b6d-9ec2-5158645a3815" />
2. The delete modal from invoice me in now in light mode, probably not intended :heavy_multiplication_x: 
<img width="1217" height="1029" alt="image" src="https://github.com/user-attachments/assets/4ec60c37-70a7-4324-abfc-86ceb1903a96" />
3.User popup is now in dark mode :heavy_check_mark: 
<img width="495" height="373" alt="image" src="https://github.com/user-attachments/assets/92f2f01a-4e8b-40b4-9964-c65b60b30de4" />
4. The `/invoices/:id` page now looks nice :heavy_check_mark: 
<img width="2442" height="1425" alt="image" src="https://github.com/user-attachments/assets/fe63ba7c-7a34-4ffb-8573-03c78bacd9f4" />

The dark mode blip doesn't happen to me anymore, nice job sorting that out!

There is one more issue similar to the blip we had before, but now it at least resolves itself.
When I load any page in dark mode initially, it shows up in a half light and half dark mode state, but resolves itself after all the data is filled in.

https://github.com/user-attachments/assets/d97e09bd-59f4-4aee-b391-aec8d40a72ad

Do you think you can solve this too? 
**Reproduction steps:** Just set your theme to dark and reload any page

Also needs a rebase, let me know if I can be of help there too :pray:

mayur1377 committed:
feat: enhance theme handling and improve background rendering

- Added `enableColorScheme` and `disableTransitionOnChange` to ThemeProvider for better theme management.
- Implemented a mounted state in BackgroundWrapper to prevent theme-dependent rendering during initial load.
- Updated background styles to use CSS variables for improved consistency across themes.
- Simplified class names in InvoiceMeLink component for cleaner styling.

mayur1377 commented:
> When I load any page in dark mode initially, it shows up in a half light and half dark mode state, but resolves itself after all the data is filled in.



> > > Alright thanks a bunch for the updates! Just got around to checking this PR out again.
> > > Let's go over the issues I've pointed out last time first:
> > > 
> > > 1. Invoice me links are now correctly styled ✔️
> > > 
> > > <img alt="image" width="1120" height="703" src="https://private-user-images.githubusercontent.com/23449297/484148437-2a532c59-ed49-4b6d-9ec2-5158645a3815.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MTU4MzQsIm5iZiI6MTc1NzUxNTUzNCwicGF0aCI6Ii8yMzQ0OTI5Ny80ODQxNDg0MzctMmE1MzJjNTktZWQ0OS00YjZkLTllYzItNTE1ODY0NWEzODE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTEwVDE0NDUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwODIzOWFhMmU5NjIzOTE0NDlkMzE2ZTdiZWUwZjYwZmNhYWVkY2U0Y2UzOWIxZjJiMmI1ZDNhMTBlNmQ3M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DVcBhx4bfHD-qxxgwstwoj80neKfGzOv8OLMQxu2Il0"> 2. The delete modal from invoice me in now in light mode, probably not intended ✖️ <img alt="image" width="1217" height="1029" src="https://private-user-images.githubusercontent.com/23449297/484148809-4ec60c37-70a7-4324-abfc-86ceb1903a96.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MTU4MzQsIm5iZiI6MTc1NzUxNTUzNCwicGF0aCI6Ii8yMzQ0OTI5Ny80ODQxNDg4MDktNGVjNjBjMzctNzBhNy00MzI0LWFiZmMtODZjZWIxOTAzYTk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTEwVDE0NDUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzYWU1MzA3ZjFjM2Y1MGMwZGM2YTE5ZDg3MTVlNzY0MmNiMmNiYmJlMGMyM2U1MWRkYzc3YmEwNjBmZmM2NzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DVJwRG85AOQnIFAxKhqvthmSmV9LVnpfyxW7lbyjD-8"> 3.User popup is now in dark mode ✔️ <img alt="image" width="495" height="373" src="https://private-user-images.githubusercontent.com/23449297/484148993-92f2f01a-4e8b-40b4-9964-c65b60b30de4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MTU4MzQsIm5iZiI6MTc1NzUxNTUzNCwicGF0aCI6Ii8yMzQ0OTI5Ny80ODQxNDg5OTMtOTJmMmYwMWEtNGU4Yi00MGI0LTk5NjQtYzY1YjYwYjMwZGU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTEwVDE0NDUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlNTcwZDdlMWRkMWY2MWQyMWEzMTEyZDQ1N2RkZDUxZWE4OGQ5ZTAzYTIzYmQwYjYzZGE1ZTNkZGE2ZDk0NDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FxYGsCviVc31V9YF8MNUYkNZjJo4Lb5M2vfYB5Lz2_0"> 4. The `/invoices/:id` page now looks nice ✔️ <img alt="image" width="2000" height="1425" src="https://private-user-images.githubusercontent.com/23449297/484149306-fe63ba7c-7a34-4ffb-8573-03c78bacd9f4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MTU4MzQsIm5iZiI6MTc1NzUxNTUzNCwicGF0aCI6Ii8yMzQ0OTI5Ny80ODQxNDkzMDYtZmU2M2JhN2MtN2EzNC00ZmZiLTg1NzMtMDNjNzhiYWNkOWY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTEwVDE0NDUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4M2QxZjYyMTMyZmYyZmFkMDgzZGQ3MDJmODdhNWRkNDZlZTBjZmNlNTBkMjkyN2JjZjU4NWZmMDhjNzczMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UOqxhzajapRMIKThYZxzCTpVAxkgBpWwNfc9iu_tar8">
> > > The dark mode blip doesn't happen to me anymore, nice job sorting that out!
> > > There is one more issue similar to the blip we had before, but now it at least resolves itself. When I load any page in dark mode initially, it shows up in a half light and half dark mode state, but resolves itself after all the data is filled in.
> > > initial-load-dark-mode-2025-09-01_11.46.41.mp4
> > > Do you think you can solve this too? **Reproduction steps:** Just set your theme to dark and reload any page
> > > Also needs a rebase, let me know if I can be of help there too 🙏
> > 
> > 
> > hello , apologies but have been really busy in the past few weeks , kindly rebase and i will take a look!
> 
> Hey @mayur1377 , apologies that it took so long to rebase this, it's been a really busy couple of weeks. I've just pushed after resolving rebase conflicts. Let me know if you are able to take a look at the remaining issues, once the last few are solved, I will gladly approve and enjoy Easy Invoice in dark mode 😎

done , kindly take a look at it!!

bassgeta reviewed (approved):
Nice going, tried everything out and all the remaining issues seem to have been fixed :100: 
Approving and we can merge this PR when the CI passes :heavy_check_mark: 
Thanks for your contribution and your patience :heart:

bassgeta reviewed (changes requested):
My bad, upon further review I've found another bug :sweat_smile: 
Navigating between pages doesn't work anymore, I am getting this error:
<img width="2560" height="1303" alt="image" src="https://github.com/user-attachments/assets/ed93414b-9cf2-4434-9387-15b91bfb3524" />

So error 1 is:
```
SES_UNCAUGHT_EXCEPTION: Error: Rendered more hooks than during the previous render.
```

And error 2 is:

```
Warning: Cannot update a component (`HotReload`) while rendering a different component (`Router`). To locate the bad setState() call inside `Router`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render Component Stack: 
    Router webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:207
    ErrorBoundaryHandler webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/error-boundary.js:113
    ErrorBoundary webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/error-boundary.js:160
    AppRouter webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:585
    ServerRoot webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/app-index.js:112
    Root webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/app-index.js:117
[<anonymous code>:1:145535](chrome://devtools/content/webconsole/%3Canonymous%20code%3E)
```

it probably has something to do with you rendering different content until everything is loaded. This didn't happen in the previous version, so the culprit is in the latest commit.

mayur1377 committed:
refactor: update layout and background rendering for improved theme handling

- Reintroduced Google Tag Manager in the layout component.
- Enhanced background rendering logic in BackgroundWrapper to maintain DOM stability and improve visibility toggling for decorative elements based on theme state.

bassgeta reviewed (changes requested):
Alright, nice job! Everything works as intended now, navigation works and everything looks exactly as it should, good stuff :ok_hand: 

There is only one minor bug which can probably be fixed easily.
The `ModeToggle` component spits out a hydration error, and it also has one initial render issue - when dark mode is selected beforehand, upon refreshing the app the switcher toggles to light mode first and then immediately goes to dark mode.


https://github.com/user-attachments/assets/6c09c35c-68dd-48c9-a4ce-e507b91fb391

Since you've done a ton of work on this already, adding some kind of placeholder to `ModeToggle` until the component is mounted (like you have done with the background wrapper), is going to be enough.

Sorry to nag you for this one last fix :pray:

mayur1377 committed:
fixed the flicker thing , to do : toggle button hydration error

bassgeta reviewed (changes requested):
Hey!
I'm still getting the switching error in the `ModeToggle` component.
1. Set your theme to dark
2. Refresh the page and observe that the toggler is initially in light mode and gets changed to dark mode later on

[Some official docs](https://github.com/pacocoursey/next-themes?tab=readme-ov-file#avoid-hydration-mismatch)

I think that what you can do for the `ModeToggle` is that if the component isn't mounted, just render an empty button for the time being (SSR and themes are hard :sweat_smile: )
```TSX
<Button
      variant="ghost"
      className="relative h-8 w-16 rounded-full bg-muted p-0 dark:bg-muted"
      aria-label="Toggle dark mode"
      disabled
    />
```

Or something that will fill the space accordingly.

mayur1377 committed:
temp fix : disable button when refresh

bassgeta reviewed (approved):
Nice one :ok_hand: 
After running through the app, I can confirm that there are no more visual glitcches and that the theme toggler does in fact display as disabled if we have dark mode enabled system wide :100: 
Thanks for you hard work on this, now let's :shipit:
